### ISSUE TITLE
Docs: [prefer-literal-enum-member] Ensure invalid examples are invalid and remove TS errors as examples

### ISSUE THREAD

**Issue by kirkwaiblinger** [typescript-eslint/typescript-eslint#10034]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

Looking through the examples in the docs for prefer-literal-enum-member, it's not very clear what behavior is actually banned by the rule, since 2 of the 5 examples in the invalid tab are invalid TS anyway, and 1 is not actually reported by the rule.

Furthermore some of the Valid examples are not valid TS either. 

We should remove examples that have invalid TS (unless to point out explicitly "already banned by TS"), and ensure that each invalid case actually does trigger the rule. 

I've annotated the docs example ([view in playground](https://typescript-eslint.io/play/#ts=5.0.4&fileType=.ts&code=MYewdgzgLgBNBOMC8MDkAVAptVBuAUPpmAK4C2MAkmAG4CGANgJYAmMA3vjDAILJxR4AGhgB6UTABqdeEzoAjBphh0IEJgHMwZYlELdxMAEI8AIjAC0MMCFiMGIAO6Y28gJ4x0AZS7H%2B7AF8RQwB5eQArTGA7NU1tXX0xCRNzKxtYTBpiGHhMAAcQeCgXGHcYKAALZXgSJV8AYX4AAz5isjyGOmKYZmL4RgFZMA0m4Iksds7u3sx%2BhkTDFMtrWxUGB2dXD29fcxQwTEcYL0woAAoARhEAJhEAZgBKMZh68AQSaMKYJjAVWK0dGA9L4AKL8a4wADUMFuSRgIIAHnlcrFwH91ACEgFCMRyFJGKwOL4eEJfEZ%2BBhsFAvIJUM8%2BLkNLUZIMfhoGvwACz06zkeSzXy%2BRZmZbpNYbEplHbcPa89akoXJEVpVb2JyS7Y%2BbhglCiCAgHQAfUZmARolJ2KIpAoAFkmBBgJh1nQDiASBAiQYJDx1uqtp4vAIZFA2d9fgBWAB0AAZI8cKm6GK5lO6Sj9ylUqLQCWxTXRJtgw9BMHQ2CAAGYZ5TAEjwXJAzyYSZdZQAGSYfQGeYLkeJzVaTY6LdZwwzXTDfQKU0LABJ2AgAqN8AEgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0Wlsfy8AhvHpNYAW1oTEEgEa90UAO5DoTSODABfENqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)):
```ts
const str = 'Test';

enum Invalid {
  A = str, // Variable assignment

  // BAD - not allowed by TS
  B = {}, // Object assignment

  // BAD - intentionally allowed by the rule
  C = `A template literal string`, // Template literal

  // BAD - not allowed by TS
  D = new Set(1, 2, 3), // Constructor in assignment

  E = 2 + 2, // Expression assignment
}

enum Valid {
  A,
  B = 'TestStr', // A regular string
  C = 4, // A number
  
  // BAD - not allowed by TS
  D = null,

  // BAD - not allowed by TS
  E = /some_regex/,
}
```

Note when fixing the template literal example, 
```ts
enum Miscellaneous {
  // Allowed by TS starting in 5.0. Should be used in the Invalid examples instead of the current Template Literal example.
  A = `A template string that interpolates ${str}`,
}
```

### Affected URL(s)

https://typescript-eslint.io/rules/prefer-literal-enum-member/#examples

### Additional Info

This issue is just to change the docs, but I think that it would also make sense to spin off another issue to check for and remove if appropriate any unnecessary complexity in the rule related to expressions that are already banned by TS.

**Comment by kirkwaiblinger** [comment]
Marking as "accepting prs" at a minimum for the invalid example that doesn't report. I figure further conversation about other things mentioned in the issue can take place on the PR itself.
